File: practica 1/sesion 3/DS_S3_C++Filters/ParteTrabajo.cpp

#include "ParteTrabajo.h"

ParteTrabajo::ParteTrabajo(int horasExtra, int costoMaterialesExtra, int costeDesplazamiento){
	setHorasExtra(horasExtra);
	setCostoMaterialesExtra(costoMaterialesExtra);
	setCosteDesplazamiento(costeDesplazamiento);
}

ParteTrabajo::ParteTrabajo(){
	
}

int ParteTrabajo::getHorasExtra() {
	return this->horasExtra;
}

void ParteTrabajo::setHorasExtra(int unHorasExtra) {
	this->horasExtra = unHorasExtra;
}

float ParteTrabajo::getCostoMaterialesExtra() {
	return this->costoMaterialesExtra;
}

void ParteTrabajo::setCostoMaterialesExtra(float unCostoMaterialesExtra) {
	this->costoMaterialesExtra = unCostoMaterialesExtra;
}

float ParteTrabajo::getCosteDesplazamiento() {
	return this->costeDesplazamiento;
}

void ParteTrabajo::setCosteDesplazamiento(float unCosteDesplazamiento) {
	this->costeDesplazamiento = unCosteDesplazamiento;
}

File: practica 1/sesion 3/DS_S3_DartFilters/ParteTrabajo.dart
import 'Trabajo.dart';

class ParteTrabajo{
  int horasExtra, costoMaterialesExtra, costeDesplazamiento;
  ParteTrabajo(int horas, int costoMateriales, int costoDesplazamiento){
    setCosteDesplazamiento(costoDesplazamiento);
    setCosteMaterialesExtra(costoMateriales);
    setHorasExtra(horas);
  }

  int getHorasExtra(){
    return this.horasExtra;
  }

  void setHorasExtra(int horas){
   this.horasExtra = horas;
  }

  int getCosteMaterialesExtra(){
    return this.costoMaterialesExtra;
  }

  void setCosteMaterialesExtra(int costoMateriales){
    this.costoMaterialesExtra = costoMateriales ;
  }

  int getCosteDesplazamiento(){
    return this.costeDesplazamiento;
  }

  void setCosteDesplazamiento(int costoDesplazamiento){
     this.costeDesplazamiento = costoDesplazamiento;
  }
}